vms/platformvm/txs/executor/proposal_tx_executor.go

```diff
@@ -363,36 +363,36 @@ func (e *ProposalTxExecutor) RewardValidatorTx(tx *txs.RewardValidatorTx) error
 		return err
 	}
 
-	stakerTx, _, err := e.OnCommitState.GetTx(stakerToReward.TxID)
+	stakerAttributes, err := e.OnCommitState.GetStakerRewardAttributes(stakerToReward.TxID)
 	if err != nil {
-		return fmt.Errorf("failed to get next removed staker tx: %w", err)
+		return fmt.Errorf("failed to get attributes for staker %d: %w", stakerToReward.TxID, err)
 	}
 
-	// Invariant: A [txs.DelegatorTx] does not also implement the
-	//            [txs.ValidatorTx] interface.
-	switch uStakerTx := stakerTx.Unsigned.(type) {
-	case txs.ValidatorTx:
-		if err := e.rewardValidatorTx(uStakerTx, stakerToReward); err != nil {
+	switch {
+	case stakerToReward.Priority.IsPermissionedValidator():
+		// Invariant: Permissioned stakers are removed by the advancement of
+		//            time and the current chain timestamp is == this staker's
+		//            EndTime. This means only permissionless stakers should be
+		//            left in the staker set.
+		return ErrShouldBePermissionlessStaker
+	case stakerToReward.Priority.IsCurrentValidator():
+		if err := e.rewardValidatorTx(stakerToReward, stakerAttributes); err != nil {
 			return err
 		}
 
 		// Handle staker lifecycle.
 		e.OnCommitState.DeleteCurrentValidator(stakerToReward)
 		e.OnAbortState.DeleteCurrentValidator(stakerToReward)
-	case txs.DelegatorTx:
-		if err := e.rewardDelegatorTx(uStakerTx, stakerToReward); err != nil {
+	case stakerToReward.Priority.IsCurrentDelegator():
+		if err := e.rewardDelegatorTx(stakerToReward, stakerAttributes); err != nil {
 			return err
 		}
 
 		// Handle staker lifecycle.
 		e.OnCommitState.DeleteCurrentDelegator(stakerToReward)
 		e.OnAbortState.DeleteCurrentDelegator(stakerToReward)
 	default:
-		// Invariant: Permissioned stakers are removed by the advancement of
-		//            time and the current chain timestamp is == this staker's
-		//            EndTime. This means only permissionless stakers should be
-		//            left in the staker set.
-		return ErrShouldBePermissionlessStaker
+		return errors.New("unexpected staker type")
```

Comment:
Should we make this an error var?

Reply:
I can actually reuse an error from `state_helpers.go`